pkg/v2/crud/crud.go

```diff
@@ -53,10 +56,30 @@ func Delete(resource *prop.Resource, path string) error {
 	if err != nil {
 		return err
 	}
+	var query *expr.Expression
 
-	return defaultTraverse(resource.RootProperty(), skipMainSchemaNamespace(resource, head), func(nav prop.Navigator) error {
-		return nav.Delete().Error()
-	})
+	if value != nil {
```

Comment:
At a high level, what's happening here is that we form a filter expression from the given value(s) and append it to the expression created from the given path. This effectively augments the path with the a filter condition that says that the element value must match _one_ of the supplied values.